common/rpc/encryption/localStoreCertProvider.go

```diff
@@ -246,62 +264,188 @@ func (s *localStoreCertProvider) getClientTLSSettings(isWorker bool) *config.Cli
 	} else if isWorker {
 		return s.legacyWorkerSettings // legacy config case when we use Frontend.Client settings
 	} else {
+		if s.tlsSettings == nil {
+			return nil
+		}
 		return &s.tlsSettings.Client // internode client case
 	}
 }
 
-func buildCAPoolFromData(caData []string) (*x509.CertPool, error) {
-	atLeastOneCert := false
-	caPool := x509.NewCertPool()
+func (s *localStoreCertProvider) Expiring(fromNow time.Duration,
+) (expiring CertExpirationMap, expired CertExpirationMap, errs []error) {
+
+	expiring = make(CertExpirationMap)
+	expired = make(CertExpirationMap)
+	errs = make([]error, 0)
+	when := time.Now().UTC().Add(fromNow)
+
+	errs = checkTLSCertForExpiration(s.FetchServerCertificate, when, expiring, expired, errs)
```

Comment:
I would prefer this to be:
```suggestion
errs = append(errs, checkTLSCertForExpiration(s.FetchServerCertificate, when, expiring, expired)...)
```

Currently it looks confusing.

Reply:
+1.